linkml_runtime/utils/yamlutils.py

```diff
@@ -93,6 +93,15 @@ def _is_empty(v: Any) -> bool:
         # TODO: Deprecate this function and migrate the python generator over to the stand alone is_empty
         return is_empty(v)
 
+    def _normalize_assignment(self, data: Any, target_class: Type) -> None:
```

Comment:
would it be possible to separate this change into another PR?

We are discussing the need for further normalization over here: https://github.com/linkml/linkml/pull/1189

All the other changes in this PR look ready to go, but I think the issue of normalization needs further discussion